src/util.rs

```diff
@@ -5,9 +5,16 @@ use core::{mem::MaybeUninit, ptr};
 /// `MaybeUninit::slice_assume_init_mut`. Every element of `slice` must have
 /// been initialized.
 #[inline(always)]
+#[allow(unused_unsafe)] // TODO(MSRV 1.65): Remove this.
 pub unsafe fn slice_assume_init_mut<T>(slice: &mut [MaybeUninit<T>]) -> &mut [T] {
-    // SAFETY: `MaybeUninit<T>` is guaranteed to be layout-compatible with `T`.
-    &mut *(slice as *mut [MaybeUninit<T>] as *mut [T])
+    let ptr: *mut [MaybeUninit<T>] = slice;
+    // SAFETY: `MaybeUninit<T>` is guaranteed to be layout-compatible with `T`
+    // and casting between two slice types of layout-compatible types is sound.
+    let ptr = ptr as *mut [T];
```

Comment:
I think `let ptr = slice as *mut [MaybeUninit<T>] as *mut [T]` would be a bit easier to understand than `let ptr: *mut [MaybeUninit<T>] = slice`.